src/main/scala/com/kinja/amqp/MessageStore.scala

```diff
@@ -1,29 +1,44 @@
 package com.kinja.amqp
 
-import java.util.UUID
-
 import com.kinja.amqp.model.Message
 import com.kinja.amqp.model.MessageConfirmation
 
 trait MessageStore {
 
+	def saveConfirmation(conf: MessageConfirmation): Unit
+	def saveMessage(msg: Message): Unit
+
 	/**
-	 * Between start and commit, the message store should make sure that
+	 * Returns the number of deleted messages
+	 */
+	def deleteMessageUponConfirm(channelId: String, deliveryTag: Long): Int
+
+	/**
+	 * Within this method, the message store should make sure that
 	 * the loaded records are not updated/deleted by any other transaction.
 	 * If it can not be guaranteed AND multiple message repeater instances exist,
 	 * same repeaters may try to resend or delete the very same messages/confirmations at a given time
 	 */
-	def startTransaction(): Unit
-	def commit(): Unit
-	def saveConfirmation(conf: MessageConfirmation): Unit
-	def saveMessage(msg: Message): Unit
+	def withLockingTransaction[T](f: => T): T
+
+	/**
+	 * Used in withLockingTransaction
```

Comment:
I would discard these comments. If you check where it's used you can see that it's used in `withLockingTransaction`.